x-pack/plugins/observability_ai_assistant/server/service/util/flush_buffer.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { repeat } from 'lodash';
+import { Observable, OperatorFunction } from 'rxjs';
+import {
+  BufferFlushEvent,
+  StreamingChatResponseEventType,
+  StreamingChatResponseEventWithoutError,
+} from '../../../common/conversation_complete';
+
+export function flushBuffer<T extends StreamingChatResponseEventWithoutError>(
+  isCloud: boolean
+): OperatorFunction<T, T | BufferFlushEvent> {
+  return (source: Observable<T>) =>
+    isCloud
+      ? new Observable<T | BufferFlushEvent>((subscriber) => {
```

Comment:
nit: if you get rid of the ternary you can return early. This makes it easy when scanning code to see this function is only relevant on cloud. Also, it removes some indentation (always nice).

```suggestion
    if(!isCloud){
      return source;
    }
    return new Observable<T | BufferFlushEvent>((subscriber) => {
```

Reply:
Agree, much better!